src/v/kafka/server/handlers/fetch.cc

```diff
@@ -263,7 +275,7 @@ static ss::future<read_result> do_read_from_ntp(
   ssx::semaphore& memory_sem,
   ssx::semaphore& memory_fetch_sem) {
     // control available memory
-    read_result::memory_units_t memory_units;
+    read_result::memory_units_t memory_units(memory_sem, memory_fetch_sem);
```

Comment:
was this a latent bug?

Reply:
Yeah, if for some reason `read_from_partition` reads data even though `skip_read` is set a seg fault will occur when we try to adjust the semaphore units in `memory_units_t`. This doesn't happen today, but one never knows what the future holds.

 This avoids this potential issue by setting the semaphore units to some reasonable default state.